Sources/PostgresNIO/New/PSQLRowStream.swift

```diff
@@ -27,14 +29,20 @@ final class PSQLRowStream: @unchecked Sendable {
     private enum BufferState {
         case streaming(buffer: CircularBuffer<DataRow>, dataSource: PSQLRowsDataSource)
         case finished(buffer: CircularBuffer<DataRow>, commandTag: String)
+        case empty
```

Comment:
we should combine those two as well. Rename `Consumed` to `StatementSummary`? Use statementSummary instead of commandTag in finished state.